internal/core/src/storage/PayloadReader.cpp

```diff
@@ -25,69 +25,80 @@
 
 namespace milvus::storage {
 
+PayloadReader::PayloadReader(milvus::FieldDataPtr fieldData)
+    : column_type_(fieldData->get_data_type()),
+      nullable_(fieldData->IsNullable()) {
+    field_data_ = fieldData;
+}
+
 PayloadReader::PayloadReader(const uint8_t* data,
                              int length,
                              DataType data_type,
                              bool nullable,
                              bool is_field_data)
     : column_type_(data_type), nullable_(nullable) {
-    auto input = std::make_shared<arrow::io::BufferReader>(data, length);
-    init(input, is_field_data);
+    init(data, length, is_field_data);
 }
 
 void
-PayloadReader::init(std::shared_ptr<arrow::io::BufferReader> input,
```

Comment:
The compatibility logic is kept in the payloadReader, this have to be handled by if-else